### DISCUSSION TITLE
Can't figure out the static assets path?

### DISCUSSION THREAD

**Discussion by damienwebdev** [vercel/vercel#5654] [Help] [answered]
I have an Angular Universal app running on vercel and I'm trying to migrate to the new functions syntax. Unfortunately, I can't seem to figure out the new pathing for the public folder where my static assets are stored.

My app: https://daffio-next-damienwebdev.graycoreio1.vercel.app

Paths I've tried
```
/assets/pwa-install-home.svg
/dist/assets/pwa-install-home.svg
/public/assets/pwa-install-home.svg
/dist/public/assets/pwa-install-home.svg
```

**Comment by damienwebdev** [comment, marked as answer]
I managed to figure this out, but have one small quirk I'm working through. Vercel tried to infer my static assets directory as `dist` which was a mistake and took me many hours to figure out.

You would need to update as below:

![Annotation 2021-01-08 101346](https://user-images.githubusercontent.com/9029654/104031008-37eca080-519a-11eb-99a9-c9653e1f5820.png)

With the following `vercel.json`

```json
{
  "version": 2,
  "functions": {
    "api/serverless.js": {
      "includeFiles": "+(server|browser)/+(main.js|index.html)",
      "runtime": "@vercel/node@1.8.5"
    }
  },
  "rewrites": [
    { "source": "/", "destination": "api/serverless.js"},
    { "source": "(.*)", "destination": "api/serverless.js" },
    { "source": "/(.*[.].+)", "destination": "/browser/$1"}
  ]
}
```

Finally, you'll need an `api/serverless.js` with the following contents:

```js
const server = require('../server/main');

module.exports = server.app();
```

**Reply by SawkaDev** [reply]
@damienwebdev  any update on this? Is this still the fix you implemented?

**Reply by damienwebdev** [reply]
Yes @Sliffcak this is still in use in prod.

**Reply by SawkaDev** [reply]
@damienwebdev  interesting, how did you determine your files were in dist folder? When I look at the source output from a vercel deployment I only see these top level directories

in localhost i have `/public/assets/fonts` 

however in vercel deployment i have `/assets.fonts`

![image](https://user-images.githubusercontent.com/19200789/211418782-dcac05ae-8e65-413f-9202-1f7e6179ccdf.png)

**Reply by SawkaDev** [reply]
@damienwebdev I got mine to work by doing the following, this is how I can access any image or font file

If your file is in `/public/assets` you just get it simply by the following:

`path.join(process.cwd(),"public","assets",FILENAME);`
